python/fledge/services/core/api/plugins/remove.py

```diff
@@ -0,0 +1,215 @@
+    """ Check notification instance state using the given rule or delivery plugin
+    """
+    notification_instances = []
+    storage_client = connect.get_storage_async()
+    configuration_mgr = ConfigurationManager(storage_client)
+    notifications = await configuration_mgr.get_category_child("Notifications")
+    if notifications:
+        for notification in notifications:
+            notification_config = await configuration_mgr._read_category_val(notification['key'])
+            name = notification_config['name']['value']
+            channel = notification_config['channel']['value']
+            rule = notification_config['rule']['value']
+            enabled = True if notification_config['enable']['value'] == 'true' else False
+            if (channel == plugin_name and enabled) or (rule == plugin_name and enabled):
+                notification_instances.append(name)
+    return notification_instances
+
+
+def purge_plugin(plugin_type: str, name: str) -> tuple:
+    _logger.info("{} plugin removal started...".format(name))
+    original_name = name
+    # Special case handling - installed directory name Vs package name
+    # For example: Plugins like http_south Vs http-south
+    name = name.replace('_', '-').lower()
+    plugin_name = 'fledge-{}-{}'.format(plugin_type, name)
+    stdout_file_path = common.create_log_file(action='remove', plugin_name=plugin_name)
+    link = "log/" + stdout_file_path.split("/")[-1]
+    get_platform = platform.platform()
+    try:
+        if 'centos' in get_platform or 'redhat' in get_platform:
+            # TODO: Add yum based package / repo check
+            cmd = "sudo yum -y remove {} > {} 2>&1".format(plugin_name, stdout_file_path)
+        else:
+            dpkg_list = os.popen('dpkg --list fledge* 2>/dev/null')
+            ls_output = dpkg_list.read()
+            _logger.debug("dpkg list output: {}".format(ls_output))
+            if len(ls_output) :
+                f = ls_output.find(plugin_name)
+                if f == -1:
+                    raise KeyError
+            else:
+                raise KeyError
+            cmd = "sudo apt -y purge {} > {} 2>&1".format(plugin_name, stdout_file_path)
+
+        code = os.system(cmd)
+        if code:
+            raise PackageError(link)
+    except KeyError:
+        # This case is for non-package installation - python plugin path will be tried first and then C
+        _logger.info("Trying removal of manually installed plugin...")
+        if plugin_type in ['notify', 'rule']:
+            plugin_type = 'notificationDelivery' if plugin_type == 'notify' else 'notificationRule'
+        try:
+            path = PYTHON_PLUGIN_PATH+'{}/{}'.format(plugin_type, original_name)
+            if os.path.isdir(path):
+                code = os.system('rm -rv {}'.format(path))
+            else:
+                path = C_PLUGINS_PATH + '{}/{}'.format(plugin_type, original_name)
+                code = os.system('rm -rv {}'.format(path))
```

Comment:
Will it not give a permission error in case of non package removal when fledge setup with `sudo make install`? With `make` it is okay.

Reply:
Yes, let's handle it in next PR.

Reply:
resolved!